### PR TITLE
kafka-clients 3.0.0 Compatibility

### PR DESCRIPTION
**Pull request by garyrussell**
The upcoming 3.0.0 release adds a new method to `Consumer` and removes
a deprecated one from `ConsumerRecord`.

Use reflection for these two methods to enable reactor-kafka to be used
with the older clients as well as 3.0.0 when used in other projects, such
as Spring for Apache Kafka.

### CODE REVIEW

File: src/test/java/reactor/kafka/mock/MockConsumer.java
Comment (artembilan) [inline src/test/java/reactor/kafka/mock/MockConsumer.java:53]: Is this Project Reactor code style to have Java imports in the end?

File: src/test/java/reactor/kafka/util/ConsumerDelegateTests.java
Comment (artembilan) [inline src/test/java/reactor/kafka/util/ConsumerDelegateTests.java:34]: Are you going to finish this test?

**Review by artembilan — CHANGES_REQUESTED** [review]
Some minor nit-picks, otherwise LGTM

File: src/test/java/reactor/kafka/util/ConsumerDelegateTests.java
Comment (garyrussell) [reply src/test/java/reactor/kafka/util/ConsumerDelegateTests.java:34]: It requires the 3.0.0 client on the classpath - I used it to test with that client - commented out for older clients.

File: src/test/java/reactor/kafka/mock/MockConsumer.java
Comment (garyrussell) [reply src/test/java/reactor/kafka/mock/MockConsumer.java:53]: Yes; this class was not consistent.

**Review by artembilan — APPROVED** [review]
LGTM, but:

> Only those with write access to this repository can merge pull requests.

So, sorry, I can't merge this for you at the moment.

**Review by simonbasle — APPROVED** [review]
sorry this one fell through the cracks. lgtm, and I'll also defer to @artembilan review 👍

**Pull request merged**
